### PR TITLE
[Bug] [Connector-V2] Fix InfluxDB SQL invalide when SQL contains tz function

### LINKED ISSUE #4231: [Bug] connector-influxdb module  InfluxDBSource class's initColumnsIndex method, sql  invalide when sql contains tz function  cause by direct  append QUERY_LIMIT
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

connector-influxdb module  InfluxDBSource class's initColumnsIndex method, sql  invalide when sql contains tz function  cause by direct  append QUERY_LIMIT 

fix code:
 `      String sql = sourceConfig.getSql();
    	String query = sql + QUERY_LIMIT;
        // if sql contains tz(), can't be append QUERY_LIMIT at last . see bug #4231
        if (sql.toLowerCase().contains("tz")) {
        	int start = sql.toLowerCase().indexOf("tz");
        	StringBuilder tmpSql = new StringBuilder(sql);
        	tmpSql.insert(start-1, QUERY_LIMIT+" ");
        	query = tmpSql.toString();
        } `
       

### SeaTunnel Version

2.3.0-release

### SeaTunnel Config

```conf
ingore
```

### Running Command

```shell
ingore
```

### Error Exception

```log
SQL syntax error
```

### Flink or Spark Version

2.4.8

### Java or Scala Version

1.8

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhengyuan-cn**
[Bug] connector-influxdb dependency influxdb-java's version shuld be 2.15 not 2.21 #4230

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by zhengyuan-cn** [comment]
[Bug] connector-influxdb module InfluxDBSource class's initColumnsIndex method, sql invalide when sql contains tz function cause by direct append QUERY_LIMIT #4231

commit id: [1d1afc5]  (https://github.com/apache/incubator-seatunnel/pull/4283/commits/1d1afc56407f093af3e6e45a4f435ac28c2ca4e5)

**Comment by Hisoka-X** [comment]
Hi, why downgraded version？IMO, we should fix a bug by upgrade version or code, not downgraded version.

**Comment by zhengyuan-cn** [comment]
hi,  version 2.21 ,It's Happened NoSuchMethod when the dependency of influxdb-java's version 2.11 .
downgraded  version  to 2.15  it works  correct.

exception detail:
23/02/27 10:50:06 INFO AbstractPluginDiscovery: Load plugin: PluginIdentifier{engineType='seatunnel', pluginType='source', pluginName='InfluxDB'} from classpath
Exception in thread "main" java.lang.NoSuchMethodError: kotlin.collections.ArraysKt.copyInto([B[BIII)[B
at shaded.okio.Segment.writeTo(Segment.kt:169)
at shaded.okio.Segment.compact(Segment.kt:152)
at shaded.okio.Buffer.write(Buffer.kt:1842)
at shaded.okio.Buffer.read(Buffer.kt:1854)
at shaded.okio.Buffer.writeAll(Buffer.kt:1642)
at shaded.okio.Options$Companion.buildTrieRecursive(Options.kt:187)
at shaded.okio.Options$Companion.buildTrieRecursive(Options.kt:174)
at shaded.okio.Options$Companion.buildTrieRecursive$default(Options.kt:113)
at shaded.okio.Options$Companion.of(Options.kt:72)
at shaded.okhttp3.internal.Util.(Util.kt:71)
at shaded.okhttp3.internal.concurrent.TaskRunner.(TaskRunner.kt:309)
at shaded.okhttp3.ConnectionPool.(ConnectionPool.kt:41)
at shaded.okhttp3.ConnectionPool.(ConnectionPool.kt:47)
at shaded.okhttp3.OkHttpClient$Builder.(OkHttpClient.kt:471)
at org.apache.seatunnel.connectors.seatunnel.influxdb.client.InfluxDBClient.getInfluxDB(InfluxDBClient.java:42)
at org.apache.seatunnel.connectors.seatunnel.influxdb.source.InfluxDBSource.prepare(InfluxDBSource.java:84)
at org.apache.seatunnel.core.starter.spark.execution.SourceExecuteProcessor.initializePlugins(SourceExecuteProcessor.java:88)
at org.apache.seatunnel.core.starter.spark.execution.AbstractPluginExecuteProcessor.(AbstractPluginExecuteProcessor.java:49)
at org.apache.seatunnel.core.starter.spark.execution.SourceExecuteProcessor.(SourceExecuteProcessor.java:51)
at org.apache.seatunnel.core.starter.spark.execution.SparkExecution.(SparkExecution.java:50)
at org.apache.seatunnel.core.starter.spark.command.SparkApiTaskExecuteCommand.execute(SparkApiTaskExecuteCommand.java:54)
at org.apache.seatunnel.core.starter.Seatunnel.run(Seatunnel.java:39)
at org.apache.seatunnel.core.starter.spark.SeatunnelSpark.main(SeatunnelSpark.java:34)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:855)
at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:161)
at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:184)
at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:86)
at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:930)
at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:939)
at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)

**Comment by Hisoka-X** [comment]
> hi, version 2.21 ,It's Happened NoSuchMethod when the dependency of influxdb-java's version 2.11 . downgraded version to 2.15 it works correct.
> 
> exception detail: 23/02/27 10:50:06 INFO AbstractPluginDiscovery: Load plugin: PluginIdentifier{engineType='seatunnel', pluginType='source', pluginName='InfluxDB'} from classpath Exception in thread "main" java.lang.NoSuchMethodError: kotlin.collections.ArraysKt.copyInto([B[BIII)[B at shaded.okio.Segment.writeTo(Segment.kt:169) at shaded.okio.Segment.compact(Segment.kt:152) at shaded.okio.Buffer.write(Buffer.kt:1842) at shaded.okio.Buffer.read(Buffer.kt:1854) at shaded.okio.Buffer.writeAll(Buffer.kt:1642) at shaded.okio.Options$Companion.buildTrieRecursive(Options.kt:187) at shaded.okio.Options$Companion.buildTrieRecursive(Options.kt:174) at shaded.okio.Options$Companion.buildTrieRecursive$default(Options.kt:113) at shaded.okio.Options$Companion.of(Options.kt:72) at shaded.okhttp3.internal.Util.(Util.kt:71) at shaded.okhttp3.internal.concurrent.TaskRunner.(TaskRunner.kt:309) at shaded.okhttp3.ConnectionPool.(ConnectionPool.kt:41) at shaded.okhttp3.ConnectionPool.(ConnectionPool.kt:47) at shaded.okhttp3.OkHttpClient$Builder.(OkHttpClient.kt:471) at org.apache.seatunnel.connectors.seatunnel.influxdb.client.InfluxDBClient.getInfluxDB(InfluxDBClient.java:42) at org.apache.seatunnel.connectors.seatunnel.influxdb.source.InfluxDBSource.prepare(InfluxDBSource.java:84) at org.apache.seatunnel.core.starter.spark.execution.SourceExecuteProcessor.initializePlugins(SourceExecuteProcessor.java:88) at org.apache.seatunnel.core.starter.spark.execution.AbstractPluginExecuteProcessor.(AbstractPluginExecuteProcessor.java:49) at org.apache.seatunnel.core.starter.spark.execution.SourceExecuteProcessor.(SourceExecuteProcessor.java:51) at org.apache.seatunnel.core.starter.spark.execution.SparkExecution.(SparkExecution.java:50) at org.apache.seatunnel.core.starter.spark.command.SparkApiTaskExecuteCommand.execute(SparkApiTaskExecuteCommand.java:54) at org.apache.seatunnel.core.starter.Seatunnel.run(Seatunnel.java:39) at org.apache.seatunnel.core.starter.spark.SeatunnelSpark.main(SeatunnelSpark.java:34) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52) at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:855) at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:161) at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:184) at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:86) at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:930) at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:939) at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)

Are we use old api to access influxDB? Can we upgrade version and modify the way access influxDB?

**Comment by zhengyuan-cn** [comment]
> > hi, version 2.21 ,It's Happened NoSuchMethod when the dependency of influxdb-java's version 2.11 . downgraded version to 2.15 it works correct.
> > exception detail: 23/02/27 10:50:06 INFO AbstractPluginDiscovery: Load plugin: PluginIdentifier{engineType='seatunnel', pluginType='source', pluginName='InfluxDB'} from classpath Exception in thread "main" java.lang.NoSuchMethodError: kotlin.collections.ArraysKt.copyInto([B[BIII)[B at shaded.okio.Segment.writeTo(Segment.kt:169) at shaded.okio.Segment.compact(Segment.kt:152) at shaded.okio.Buffer.write(Buffer.kt:1842) at shaded.okio.Buffer.read(Buffer.kt:1854) at shaded.okio.Buffer.writeAll(Buffer.kt:1642) at shaded.okio.Options$Companion.buildTrieRecursive(Options.kt:187) at shaded.okio.Options$Companion.buildTrieRecursive(Options.kt:174) at shaded.okio.Options$Companion.buildTrieRecursive$default(Options.kt:113) at shaded.okio.Options$Companion.of(Options.kt:72) at shaded.okhttp3.internal.Util.(Util.kt:71) at shaded.okhttp3.internal.concurrent.TaskRunner.(TaskRunner.kt:309) at shaded.okhttp3.ConnectionPool.(ConnectionPool.kt:41) at shaded.okhttp3.ConnectionPool.(ConnectionPool.kt:47) at shaded.okhttp3.OkHttpClient$Builder.(OkHttpClient.kt:471) at org.apache.seatunnel.connectors.seatunnel.influxdb.client.InfluxDBClient.getInfluxDB(InfluxDBClient.java:42) at org.apache.seatunnel.connectors.seatunnel.influxdb.source.InfluxDBSource.prepare(InfluxDBSource.java:84) at org.apache.seatunnel.core.starter.spark.execution.SourceExecuteProcessor.initializePlugins(SourceExecuteProcessor.java:88) at org.apache.seatunnel.core.starter.spark.execution.AbstractPluginExecuteProcessor.(AbstractPluginExecuteProcessor.java:49) at org.apache.seatunnel.core.starter.spark.execution.SourceExecuteProcessor.(SourceExecuteProcessor.java:51) at org.apache.seatunnel.core.starter.spark.execution.SparkExecution.(SparkExecution.java:50) at org.apache.seatunnel.core.starter.spark.command.SparkApiTaskExecuteCommand.execute(SparkApiTaskExecuteCommand.java:54) at org.apache.seatunnel.core.starter.Seatunnel.run(Seatunnel.java:39) at org.apache.seatunnel.core.starter.spark.SeatunnelSpark.main(SeatunnelSpark.java:34) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52) at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:855) at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:161) at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:184) at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:86) at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:930) at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:939) at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
> 
> Are we use old api to access influxDB? Can we upgrade version and modify the way access influxDB?

I use influxDB version 1.8.10,  seatunnel-influxDB  use simple config. as below
`InfluxDB { 
    url="http://10.x.x.22:8086"
    database="data_storage"
    password="xxxxxx"
    sql="select temp3,time from lunzi where time >= '2023-01-11 16:18:00' and time < '2023-01-11 16:19:00' tz('Asia/Shanghai')"
    fields { 
                temp3 = float
                time = bigint
        } 

    result_table_name="source1"
}`

**Comment by zhengyuan-cn** [comment]
> > hi, version 2.21 ,It's Happened NoSuchMethod when the dependency of influxdb-java's version 2.11 . downgraded version to 2.15 it works correct.
> > exception detail: 23/02/27 10:50:06 INFO AbstractPluginDiscovery: Load plugin: PluginIdentifier{engineType='seatunnel', pluginType='source', pluginName='InfluxDB'} from classpath Exception in thread "main" java.lang.NoSuchMethodError: kotlin.collections.ArraysKt.copyInto([B[BIII)[B at shaded.okio.Segment.writeTo(Segment.kt:169) at shaded.okio.Segment.compact(Segment.kt:152) at shaded.okio.Buffer.write(Buffer.kt:1842) at shaded.okio.Buffer.read(Buffer.kt:1854) at shaded.okio.Buffer.writeAll(Buffer.kt:1642) at shaded.okio.Options$Companion.buildTrieRecursive(Options.kt:187) at shaded.okio.Options$Companion.buildTrieRecursive(Options.kt:174) at shaded.okio.Options$Companion.buildTrieRecursive$default(Options.kt:113) at shaded.okio.Options$Companion.of(Options.kt:72) at shaded.okhttp3.internal.Util.(Util.kt:71) at shaded.okhttp3.internal.concurrent.TaskRunner.(TaskRunner.kt:309) at shaded.okhttp3.ConnectionPool.(ConnectionPool.kt:41) at shaded.okhttp3.ConnectionPool.(ConnectionPool.kt:47) at shaded.okhttp3.OkHttpClient$Builder.(OkHttpClient.kt:471) at org.apache.seatunnel.connectors.seatunnel.influxdb.client.InfluxDBClient.getInfluxDB(InfluxDBClient.java:42) at org.apache.seatunnel.connectors.seatunnel.influxdb.source.InfluxDBSource.prepare(InfluxDBSource.java:84) at org.apache.seatunnel.core.starter.spark.execution.SourceExecuteProcessor.initializePlugins(SourceExecuteProcessor.java:88) at org.apache.seatunnel.core.starter.spark.execution.AbstractPluginExecuteProcessor.(AbstractPluginExecuteProcessor.java:49) at org.apache.seatunnel.core.starter.spark.execution.SourceExecuteProcessor.(SourceExecuteProcessor.java:51) at org.apache.seatunnel.core.starter.spark.execution.SparkExecution.(SparkExecution.java:50) at org.apache.seatunnel.core.starter.spark.command.SparkApiTaskExecuteCommand.execute(SparkApiTaskExecuteCommand.java:54) at org.apache.seatunnel.core.starter.Seatunnel.run(Seatunnel.java:39) at org.apache.seatunnel.core.starter.spark.SeatunnelSpark.main(SeatunnelSpark.java:34) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52) at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:855) at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:161) at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:184) at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:86) at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:930) at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:939) at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
> 
> Are we use old api to access influxDB? Can we upgrade version and modify the way access influxDB?

I tested influxdb-java version 2.21-2.23 them not support, the Exception still .

**Comment by Hisoka-X** [comment]
> I tested influxdb-java version 2.21-2.23 them not support, the Exception still .

Maybe we should update the code which SeaTunnel access InfluxDB.

**Comment by zhengyuan-cn** [comment]
> > I tested influxdb-java version 2.21-2.23 them not support, the Exception still .
> 
> Maybe we should update the code which SeaTunnel access InfluxDB.
hi, code in this commit, ,for bug #4231
[Bug] connector-influxdb module InfluxDBSource class's initColumnsIndex method, sql invalide when sql contains tz function cause by direct append QUERY_LIMIT #4231

**Comment by zhengyuan-cn** [comment]
The failure check (Backend / unit-test (11, ubuntu-latest) need resolve? how can I resolve it?

**Comment by Hisoka-X** [comment]
Try merge from dev.

File: seatunnel-connectors-v2/connector-influxdb/pom.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@
     <name>SeaTunnel : Connectors V2 : Influxdb</name>
 
     <properties>
-        <influxdb.version>2.21</influxdb.version>
+        <influxdb.version>2.15</influxdb.version>
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-influxdb/pom.xml]: Do not downgrade version.

Comment (zhengyuan-cn) [reply seatunnel-connectors-v2/connector-influxdb/pom.xml]: ok,  let me recovery to 2.21

**Comment by Hisoka-X** [comment]
Hi, can you help add this test case (SQL contains tz function) to InfluxDB E2E test?

**Comment by zhengyuan-cn** [comment]
> 

Ok

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-influxdb-e2e/src/test/java/org/apache/seatunnel/e2e/connector/influxdb/InfluxdbIT.java [hunk 2]
```diff
@@ -207,6 +207,38 @@ public void testInfluxdb(TestContainer container) throws IOException, Interrupte
             }
         }
     }
+    
+    @TestTemplate
+    public void testInfluxdbSqlTest(TestContainer container) throws IOException, InterruptedException {
+        Container.ExecResult execResult = container.executeJob("/influxdb-to-influxdb-sqltest.conf");
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-influxdb-e2e/src/test/java/org/apache/seatunnel/e2e/connector/influxdb/InfluxdbIT.java]: un commit this file？

influxdb-to-influxdb-sqltest.conf

Comment (hailin0) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-influxdb-e2e/src/test/java/org/apache/seatunnel/e2e/connector/influxdb/InfluxdbIT.java]: @zhengyuan-cn

**Comment by hailin0** [comment]
check ci error

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-influxdb-e2e/src/test/java/org/apache/seatunnel/e2e/connector/influxdb/InfluxdbIT.java (hunk 2 above)
Comment (zhengyuan-cn) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-influxdb-e2e/src/test/java/org/apache/seatunnel/e2e/connector/influxdb/InfluxdbIT.java]: sorry I forgot commit it,

Now,   commited

**Comment by zhengyuan-cn** [comment]
recommited RP see: https://github.com/apache/seatunnel/pull/4829

**Pull request closed without merging**